main.tf

```diff
@@ -1,28 +1,3 @@
-data "aws_iam_policy_document" "ssm_maintenance_window" {
-  statement {
-    actions = ["sts:AssumeRole"]
-    principals {
-      type = "Service"
-      identifiers = [
-        "ec2.amazonaws.com",
-        "ssm.amazonaws.com"
-      ]
-    }
-  }
-}
-
-resource "aws_iam_role" "ssm_maintenance_window" {
-  name_prefix        = var.name
-  assume_role_policy = data.aws_iam_policy_document.ssm_maintenance_window.json
-  path               = "/system/"
-  tags               = var.tags
-}
-
-resource "aws_iam_role_policy_attachment" "role_attach_ssm_mw" {
-  role       = aws_iam_role.ssm_maintenance_window.name
-  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonSSMMaintenanceWindowRole"
-}
-
```

Comment:
You're going to have to explain why the role is getting removed.

Reply:
v3.28.0 of the provider makes it optional because AWS can automatically handle it https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.28.0